### DISCUSSION TITLE
cannot get environment values for polybar config for font size

### DISCUSSION THREAD

**Discussion by marsjane** [polybar/polybar#3056] [Q&A]
I am wondering to set different bar font size for different computers, so in the launch.sh script I am using sth like:
```
net=$(ip -o -4 route show to default | awk '{print $5}')
p_fontsize=20
if [ "$my_hostname" = "my-pc" ]; then
  p_fontsize=16
fi

if type "xrandr" > /dev/null; then
  for m in $(xrandr --query | grep " connected" | cut -d" " -f1); do
    PFONTSIZE=$p_fontsize NETINTERFACE=$net MONITOR=$m polybar --reload mybar -c ~/.config/i3/polybar/config &
  done
else
  polybar --reload mybar -c ~/.config/i3/polybar/config &
fi

```
and in the ~/.config/i3/polybar/config file i use
```
font-0 = fixed:pixelsize=${env:PFONTSIZE};1
font-1 = unifont:fontformat=truetype:size=10:antialias=false;0
font-2 = siji:pixelsize=16;1
```
but it seems the ${env:PFONTSIZE} didn't make any difference when I change the value in the launch.sh. So the env value not successfully imported.(If I modify it manually in the config file it will work). Also, the NETINTERFACE env works fine when I use it in the config file with monitor = ${env:MONITOR}, so what is wrong? 
Can anyone give some advice?

**Comment by patrick96** [comment]
In the polybar config, you can't mix the `${...}` references with other things, only `key = ${...}` works, `key = text ${...} some more text` does not. See also the [docs](https://github.com/polybar/polybar/wiki/Configuration#referencing-values).

What I do in my config is run this in the bash script that starts my bar: `export BAR_FONT1="Symbols Nerd Font Mono:style=2048-em:size=${FONT_SIZE}"`. And use it as `font-1 = ${env:BAR_FONT1}`
